### ISSUE TITLE
Can not install with Windows 11 ARM

### ISSUE THREAD

**Issue by codezi** [ramensoftware/windhawk#11]
Hello, I am using Windows 11 ARM on Parallels for M1 Macbook.
And I can not install Windhawk.

**Comment by m417z** [comment]
Hi,
Unfortunately, Windhawk doesn't support ARM at the moment. Only x86 and x86-64 architectures are supported.

**Comment by codezi** [comment]
Thank you. I hope that it supports ARM soon. I like your Windhank very much!!!

**Comment by m417z** [comment]
No such plan for the near future, sorry. The main obstacle for making Windhawk work on ARM is that [MinHook](https://github.com/TsudaKageyu/minhook) only supports x86/x64. I thought of trying to use [Detours](https://github.com/microsoft/Detours) for ARM, but adapting it will require some work. Also I don't currently have an ARM device for development and testing, and getting one is not a priority.

**Comment by Sandritsch91** [comment]
I also would really appreciate arm support for this great application.

Keep up the good work!

**Comment by ComputerFields** [comment]
I just purchased an ARM64 powered MS Surface Pro 11 Copilot+ PC and would very much like to use Windhawk.

OS Name	Microsoft Windows 11 Pro
Version	10.0.26100 Build 26100
Other OS Description 	Not Available
OS Manufacturer	Microsoft Corporation
System Manufacturer	Microsoft Corporation
System Model	Microsoft Surface Pro, 11th Edition
System Type	ARM64-based PC
System SKU	Surface_Pro_11th_Edition_2076
Processor	Snapdragon(R) X 12-core X1E80100 @ 3.40 GHz, 4012 Mhz, 12 Core(s), 12 Logical Processor(s)
BIOS Version/Date	Microsoft Corporation 144.2019.235, 6/13/2024

**Comment by adamlogan** [comment]
Also wanting to use on M1 MBP via Parallels. Bummed Windhawk doesn't work for this architecture. Would appreciate it if you added something to the installer to check for ARM architecture and stop with a notification letting us know until you do make it ARM compatible. Would be good to state it is only x86 and x86-64 architecture supported on the download page or something.

**Comment by afit** [comment]
> Also I don't currently have an ARM device for development and testing, and getting one is not a priority.

If this becomes a priority I could ship you an Arm device or, if more convenient for you, could get you remote admin access to one.

**Comment by m417z** [comment]
Thanks @afit, remote access to an ARM machine would be a good start. I'll let you know once I'll have some time to work on it.

**Comment by tristratos** [comment]
Yes please add windows arm support!!!

I am also running windows on arm through MacBook m3 device and seems installer refuses to install windhawk as windhawk service does not run at all for some reason.

Please take a look at it. It should be probably something easy to fix 1

**Comment by msubodh** [comment]
Appreciate the great work. Just adding my +1 to this

**Comment by Okaza03** [comment]
Amazing work. Adding a +1 to this.

**Comment by tristratos** [comment]
Any news / progress on this?
Should we expect this to come soon ?

**Comment by Test-Acount** [comment]
same here. win11arm can't start the windhawk service. thx.

**Comment by JH3003** [comment]
Hi, I also use MS Copilot 11 Pro with ARM and it is amazing! It would be great if WindHawk can support also this Windows Version. Do not having Taskbar on Top is still anoying.

**Comment by Hidlsh** [comment]
Just bought a MS Surface Laptop 7 with ARM, and just realized windhawk doesn't support it yet still. Please let us know if you plan on pursuing ARM support or conversion any time soon.

**Comment by Snazzie** [comment]
+1 from lenovo snapdragon elite 7x user.

**Comment by JonathanSCE** [comment]
+1 from a Microsoft Surface Pro 11th Edition.

**Comment by breathless19** [comment]
+1 from Asus Vivobook S 15 Plus.

**Comment by TiagoRemedio** [comment]
+1 from Surface Pro 11th Edition.

**Comment by Alonely0** [comment]
+1 from Surface Pro 11th Edition, willing to help with dev or testing.

**Comment by BitsofBits2435** [comment]
+1 from the Samsung Galalxy EdgeBook 4 Edge. Would love ARM WIndHawk.

**Comment by Imacolombian346** [comment]
+1 from Windows from parallels for Mac Mini m4. this thing is a BEAST, and i would love to have WindHawk for ARM.
PD: if any of you need any tester for making this real, im willing to help 🚀🚀

**Comment by BitsofBits2435** [comment]
Thought i'd jump back in. 
If a test device is needed that nativley runs windows (no parallels or anything like that), I'd be more than happy to offer my Galaxy Book 4Edge for testing.

**Comment by Alonely0** [comment]
I'm really busy IRL and it doesn't seem like it'll get much better until late-June, but I'm going to try to work on a Detours port of the engine on my free time. It seems like a big hassle since the API surface of Detours is orders of magnitude larger than Minhook's (its name checks out, huh), but I've looked at it and I have some ideas. I'm also not experienced in C++/C# Windows development at all, but it is what it is.

**Comment by m417z** [comment]
@Alonely0 sure, any progress towards ARM64 compatibility is welcome and can help. Although I haven't looked at it enough, I'm afraid that creating a Detours wrapper for the MinHook API isn't going to be possible without some changes to the Detours internals. One example is the lack of support (AFAIK) for placing multiple hooks on the same function in Detours.

**Comment by purple-applepie** [comment]
I'm on an XPS 9345 and I would really like ARM64 support!

**Comment by Flowgem** [comment]
+ 1 From XPS 13 AI

**Comment by rexxDZN** [comment]
+ 1 cant use it at all under emulation

**Comment by ArjiAmin** [comment]
+1 From an omnibook, and i am 100% willing to install and test if everything is working correctly if there's ever a beta for the arm64 architecture.

**Comment by akamienski** [comment]
Is there any update on this? I would love to purchase ARM laptop, but lack of arm support makes me reluctant to it.

**Comment by OrthodoxWindows** [comment]
> Is there any update on this? I would love to purchase ARM laptop, but lack of arm support makes me reluctant to it.

It is absolutely not a solution, but there are AMD64 devices that may look like ARM devices. In particular the Microsoft Surfaceptro, and the Lenovo Yoga.
I imagine that there can be performance advantages in an ARM device, but there are AMD64 devices that are very "compact" ergonomics.

**Comment by m417z** [comment]
Indeed, ARM64 support is finally available in Windhawk v1.6! Thank you for your feedback and patience.

Here's the journey of adding ARM64 support in technical terms, summarized in a meme:

![Image](https://github.com/user-attachments/assets/aee62308-8ce6-4e7a-be7d-e25c2e989e8f)

**Comment by rexxDZN** [comment]
thank you very much. I can not only launch the exe to download Windhawk now. But also run it. 
One thing, is it normal more than 5 sec wait time while it says installing when downloading a mod? or does the app just need more optimization to arm.

**Comment by m417z** [comment]
> is it normal more than 5 sec wait time while it says installing when downloading a mod?

Yes, it's normal, compilation can take a long time, depending on the mod. I plan on adding an option to download a precompiled mod instead, which should be much faster, in a future Windhawk update.

**Comment by kristibektashi** [comment]
> Indeed, ARM64 support is finally available in Windhawk v1.6! Thank you for your feedback and patience.
> 
> Here's the journey of adding ARM64 support in technical terms, summarized in a meme:
> 
> ![Image](https://github.com/user-attachments/assets/aee62308-8ce6-4e7a-be7d-e25c2e989e8f)

Is hooking ARM32 processes on ARM64 versions of Windows 10 and 11 prior to 24H2 supported? @m417z
